src/eval-frame/components/reps/__test_helpers__/reps-test-value-cases.js

```diff
@@ -0,0 +1,283 @@
+  map: new Map([[1, "one"], [2, "two"], [3, "three"]]),
+  map_empty: new Map(),
+  set: new Set([1, 2, 3, 4, 5]),
+  set_empty: new Set(),
+  weakSet_empty: new WeakSet(),
+  weakMap_empty: new WeakMap()
+};
+
+// ==================== arrays of base objects
+
+export const arraysOfBaseObjects = {};
+Object.keys(baseObjects).forEach(k => {
+  arraysOfBaseObjects[`arrayOfObj_${k}`] = new Array(1000).fill(baseObjects[k]);
+});
+
+// ==================== composite objects
+
+export const compositeObjects = {
+  object_plainComposite1: {
+    id: 2,
+    name: "An ice sculpture",
+    price: 12.5,
+    tags: ["cold", "ice"],
+    time: new Date("2005-04-03"),
+    description: "a very long description string ... ".repeat(1000),
+    dimensions: {
+      length: 7.0,
+      width: 12.0,
+      height: 9.5
+    },
+    warehouseLocation: {
+      latitude: -78.75,
+      longitude: 20.4
+    }
+  },
+
+  object_plainComposite2: {
+    glossary: {
+      title: "example glossary",
+      GlossDiv: {
+        title: "S",
+        GlossList: {
+          GlossEntry: {
+            ID: "SGML",
+            SortAs: "SGML",
+            GlossTerm: "Standard Generalized Markup Language",
+            Acronym: "SGML",
+            Abbrev: "ISO 8879:1986",
+            GlossDef: {
+              para:
+                "A meta-markup language, used to create markup languages such as DocBook.",
+              GlossSeeAlso: ["GML", "XML"]
+            },
+            GlossSee: "markup"
+          }
+        }
+      }
+    }
+  },
+
+  object_plainComposite3: {
+    a1: 1,
+    a2: "A2",
+    a3: true,
+    a4: undefined,
+    a5: {
+      "a5-1": null,
+      "a5-2": ["a5-2-1", "a5-2-2"],
+      "a5-3": {}
+    },
+    a6: () => {
+      console.log("hello world");
+    },
+    a7: new Date("2005-04-03")
+  }
+};
+
+// ==================== custom classes
+
+class Animal {
+  constructor(name) {
+    this.name = name;
+  }
+  speak() {
+    console.log(`${this.name} makes an animal noise.`);
+  }
+}
+
+class Dog extends Animal {
+  constructor(name) {
+    super(`${name} the dog`);
+  }
+
+  speak() {
+    console.log(`${this.name} makes an animal noise.`);
+  }
+}
+
+export const customClassCases = {
+  class_1: new Animal("Bob"),
+  class_subclass: new Dog("Fido")
+};
+
+// ==================== built-ins
+
+export const builtInObjectCases = {
+  intrinsic_math: Math,
+  intrinsic_json: JSON
+};
+
+// ==================== Promise
+
+export const promiseCases = {
+  promise: new Promise(() => "resolve", () => "reject"),
+  promise_reject: Promise.reject(),
+  promise_resolve: Promise.resolve()
+};
+
+// ==================== tabular
+
+export const rowTableCases = {
+  rowsTable_plainObjects: new Array(100).fill(0).map((x, i) => ({
+    index: i,
+    time: new Date(),
+    id: Math.sin(i)
+  })),
+  rowsTable_compositeObjects: new Array(100).fill(compositeObjects),
```

Comment:
Could we actually randomly generate some of the values in `compositeObjects` so it's not a big table that never changes when I paginate etc.? I'm not very bright so I had to go check to see if it was doing the thing i wanted it to do.

Reply:
i'm all for adding more cases. i think it's generally considered wise to avoid random inputs in tests (unless you are specifically doing fuzzing), but if you want to send me a few more deterministic recipes, i'll def pop them in. the more cases we have here, the better!

Reply:
and of course, the idea is to keep adding more test values as we go, so this list need never be considered "done". b/c this is kind of a permanent work-in-progress, i'm going to mark this thread resolved, but do feel free to send me recipes and i'll pop them in if you get them to me while i'm working on this